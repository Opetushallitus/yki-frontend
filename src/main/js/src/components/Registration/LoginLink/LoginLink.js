import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from '../../../axios';
import * as Yup from 'yup';
import { withTranslation } from 'react-i18next';
import { Formik, Form, Field, ErrorMessage } from 'formik';

import classes from './LoginLink.module.css';
import Button from '../../../components/UI/Button/Button';
import Alert from '../../../components/Alert/Alert';

export class LoginLink extends Component {
  state = {
    submitted: false,
    email: null,
  };

  render() {
    const submit = payload => {
      return axios.post('/yki/api/login-link', payload);
    };

    return !this.state.submitted ? (
      <Formik
        initialValues={{ email: '' }}
        validationSchema={Yup.object().shape({
          email: Yup.string()
            .required(this.props.t('error.mandatory'))
            .email(this.props.t('error.email')),
        })}
        onSubmit={(values, { setSubmitting, setStatus }) => {
          submit({
            email: values.email,
            exam_session_id: this.props.examSessionId,
          })
            .then(() => {
              setSubmitting(false);
              setStatus(null);
              this.setState({ submitted: true, email: values.email });
            })
            .catch(err => {
              setSubmitting(false);
              setStatus(err);
            });
        }}
      >
        {({ isSubmitting, status, isValid }) => (
          <Form className={classes.Form}>
            <label htmlFor="email" className={classes.Label}>
              {this.props.t('registration.loginlink.header')}
            </label>
            <div className={classes.FormElement}>
              <Field
                type="email"
                name="email"
                className={classes.TextInput}
                data-cy="input-email"
              />
              <ErrorMessage
                name="email"
                component="span"
                className={classes.ValidationError}
              />
            </div>
            <Button
              type="submit"
              isRegistration={true}
              disabled={!isValid || isSubmitting}
              datacy="button-loginlink-form-submit"
            >
              {this.props.t('registration.notification.signup.button')}
            </Button>
            {!!status && (
              <div className={classes.SubmitError}>
                <Alert
                  title={this.props.t('error.email.sendFailed')}
                  optionalText={this.props.t('error.generic.info')}
                />
              </div>
            )}
          </Form>
        )}
      </Formik>
    ) : (
      <>
        <p data-cy="loginlink-success">
          {this.props.t('registration.loginlink.success')}{' '}
          <strong>{this.state.email}</strong>
        </p>
      </>
    );
  }
}

LoginLink.propTypes = {
  examSessionId: PropTypes.number.isRequired,
};

export default withTranslation()(LoginLink);
