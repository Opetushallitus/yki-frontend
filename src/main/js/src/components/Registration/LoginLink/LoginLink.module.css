.TextInput {
  height: 30px;
  width: 260px;
  background: hsla(0, 0%, 100%, 1) 0 0 no-repeat padding-box;
  border: 1px solid hsla(194, 82%, 43%, 1);
  border-radius: 4px;
}

.Label {
  font: normal normal 14px/24px PT Sans;
  letter-spacing: 0;
  color: hsla(194, 91%, 21%, 1);
}

.Form {
  margin-left: 5vw;
}

.Form button {
  float: left;
  margin-top: 15px;
}

.ValidationError {
  display: block;
  color: hsla(15, 100%, 40%, 1);
  margin: 14px 0px 14px 0px;
}

.SubmitError {
  display: inline-block;
  margin: 14px 0px 14px 0px;
}


