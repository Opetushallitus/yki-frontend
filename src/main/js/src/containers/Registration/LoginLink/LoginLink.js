import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from '../../../axios';
import * as Yup from 'yup';
import { withNamespaces } from 'react-i18next';
import { Formik, Form, Field, ErrorMessage } from 'formik';

import classes from './LoginLink.module.css';
import Button from '../../../components/UI/Button/Button';
import Alert from '../../../components/Alert/Alert';

export class LoginLink extends Component {
  state = {
    submitted: false,
  };

  render() {
    const submit = payload => {
      return axios.post('/yki/api/login-link', payload);
    };

    return !this.state.submitted ? (
      <Formik
        initialValues={{ email: '' }}
        validationSchema={Yup.object().shape({
          email: Yup.string()
            .required(this.props.t('error.mandatory'))
            .email(this.props.t('error.email')),
        })}
        onSubmit={(values, { setSubmitting, setStatus }) => {
          submit({
            email: values.email,
            exam_session_id: this.props.examSessionId,
          })
            .then(() => {
              setSubmitting(false);
              setStatus(null);
              this.setState({ submitted: true });
            })
            .catch(err => {
              setSubmitting(false);
              setStatus(err);
            });
        }}
      >
        {({ isSubmitting, status, isValid }) => (
          <Form className={classes.Form}>
           {!!status && <Alert title={this.props.t('error.email.sendFailed')} optionalText={this.props.t('error.generic.info')}/>}
            <h3>{this.props.t('registration.loginlink.header')}</h3>
            <div className={classes.FormElement}>
              <Field type="email" name="email" className={classes.TextInput} />
              <ErrorMessage name="email" component="span" />
            </div>
            <Button
              type="submit"
              isRegistration={true}
              disabled={!isValid || isSubmitting}
            >
              {this.props.t('examSession.addButton')}
            </Button>
          </Form>
        )}
      </Formik>
    ) : (
      <React.Fragment>
        <p>{this.props.t('registration.loginlink.success')}</p>
      </React.Fragment>
    );
  }
}

LoginLink.propTypes = {
  examSessionId: PropTypes.number.isRequired,
};

export default withNamespaces()(LoginLink);
