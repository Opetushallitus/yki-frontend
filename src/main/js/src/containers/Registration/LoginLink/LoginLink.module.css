.TextInput {
  width: 340px;
  height: 35px;
}

.TextArea {
  width: 340px;
  height: 50px;
  resize: none;
}

.Form h2 {
  font-size: 1.25em;
  line-height: 28px;
  margin: 8px 0px 14px 0px;
}

.Form h3 {
  margin: 0px 0px 8px 0px;
  line-height: 19px;
}

.Form label {
  font-size: 0.875em;
}

.Form input:focus {
  outline: 1px solid hsla(195, 100%, 47%, 1);
}

.Form button {
  float: left;
  margin-top: 15px;
}

.ErrorMessage {
  display: block;
  color: hsla(15, 100%, 40%, 1);
  margin: 14px 0px 14px 0px;
}
