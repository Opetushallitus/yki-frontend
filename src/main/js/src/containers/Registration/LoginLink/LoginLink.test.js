import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { LoginLink } from './LoginLink';

configure({ adapter: new Adapter() });


describe('<LoginLink />', () => {
  it('should render login link page', () => {
    const wrapper = mount(
      <LoginLink
        examSessionId={1}
        t={t => t}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
