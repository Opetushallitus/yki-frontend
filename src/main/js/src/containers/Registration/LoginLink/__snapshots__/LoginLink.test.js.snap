// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginLink /> should render login link page 1`] = `
<LoginLink
  examSessionId={1}
  t={[Function]}
>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "email": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": false,
        "_nodes": Array [
          "email",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
              "undefined": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    <Form
      className="Form"
    >
      <Component
        className="Form"
        formik={
          Object {
            "dirty": false,
            "errors": Object {},
            "handleBlur": [Function],
            "handleChange": [Function],
            "handleReset": [Function],
            "handleSubmit": [Function],
            "initialValues": Object {
              "email": "",
            },
            "isSubmitting": false,
            "isValid": false,
            "isValidating": false,
            "registerField": [Function],
            "resetForm": [Function],
            "setError": [Function],
            "setErrors": [Function],
            "setFieldError": [Function],
            "setFieldTouched": [Function],
            "setFieldValue": [Function],
            "setFormikState": [Function],
            "setStatus": [Function],
            "setSubmitting": [Function],
            "setTouched": [Function],
            "setValues": [Function],
            "submitCount": 0,
            "submitForm": [Function],
            "touched": Object {},
            "unregisterField": [Function],
            "validate": undefined,
            "validateField": [Function],
            "validateForm": [Function],
            "validateOnBlur": true,
            "validateOnChange": true,
            "validationSchema": ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": false,
              "_nodes": Array [
                "email",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "email": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                    "undefined": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            },
            "values": Object {
              "email": "",
            },
          }
        }
      >
        <form
          className="Form"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <h3>
            registration.loginlink.header
          </h3>
          <div
            className="FormElement"
          >
            <FormikConnect(FieldInner)
              className="TextInput"
              name="email"
              type="email"
            >
              <FieldInner
                className="TextInput"
                formik={
                  Object {
                    "dirty": false,
                    "errors": Object {},
                    "handleBlur": [Function],
                    "handleChange": [Function],
                    "handleReset": [Function],
                    "handleSubmit": [Function],
                    "initialValues": Object {
                      "email": "",
                    },
                    "isSubmitting": false,
                    "isValid": false,
                    "isValidating": false,
                    "registerField": [Function],
                    "resetForm": [Function],
                    "setError": [Function],
                    "setErrors": [Function],
                    "setFieldError": [Function],
                    "setFieldTouched": [Function],
                    "setFieldValue": [Function],
                    "setFormikState": [Function],
                    "setStatus": [Function],
                    "setSubmitting": [Function],
                    "setTouched": [Function],
                    "setValues": [Function],
                    "submitCount": 0,
                    "submitForm": [Function],
                    "touched": Object {},
                    "unregisterField": [Function],
                    "validate": undefined,
                    "validateField": [Function],
                    "validateForm": [Function],
                    "validateOnBlur": true,
                    "validateOnChange": true,
                    "validationSchema": ObjectSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_defaultDefault": [Function],
                      "_deps": Array [],
                      "_excludedEdges": Array [],
                      "_exclusive": Object {},
                      "_mutate": false,
                      "_nodes": Array [
                        "email",
                      ],
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "object",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "fields": Object {
                        "email": StringSchema {
                          "_blacklist": RefSet {
                            "list": Set {},
                            "refs": Map {},
                          },
                          "_conditions": Array [],
                          "_deps": Array [],
                          "_exclusive": Object {
                            "required": false,
                            "undefined": false,
                          },
                          "_mutate": false,
                          "_options": Object {
                            "abortEarly": true,
                            "recursive": true,
                          },
                          "_type": "string",
                          "_typeError": [Function],
                          "_whitelist": RefSet {
                            "list": Set {},
                            "refs": Map {},
                          },
                          "tests": Array [
                            [Function],
                            [Function],
                            [Function],
                          ],
                          "transforms": Array [
                            [Function],
                          ],
                        },
                      },
                      "tests": Array [],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "values": Object {
                      "email": "",
                    },
                  }
                }
                name="email"
                type="email"
              >
                <input
                  className="TextInput"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="email"
                  value=""
                />
              </FieldInner>
            </FormikConnect(FieldInner)>
            <FormikConnect(ErrorMessageImpl)
              component="span"
              name="email"
            >
              <ErrorMessageImpl
                component="span"
                formik={
                  Object {
                    "dirty": false,
                    "errors": Object {},
                    "handleBlur": [Function],
                    "handleChange": [Function],
                    "handleReset": [Function],
                    "handleSubmit": [Function],
                    "initialValues": Object {
                      "email": "",
                    },
                    "isSubmitting": false,
                    "isValid": false,
                    "isValidating": false,
                    "registerField": [Function],
                    "resetForm": [Function],
                    "setError": [Function],
                    "setErrors": [Function],
                    "setFieldError": [Function],
                    "setFieldTouched": [Function],
                    "setFieldValue": [Function],
                    "setFormikState": [Function],
                    "setStatus": [Function],
                    "setSubmitting": [Function],
                    "setTouched": [Function],
                    "setValues": [Function],
                    "submitCount": 0,
                    "submitForm": [Function],
                    "touched": Object {},
                    "unregisterField": [Function],
                    "validate": undefined,
                    "validateField": [Function],
                    "validateForm": [Function],
                    "validateOnBlur": true,
                    "validateOnChange": true,
                    "validationSchema": ObjectSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_defaultDefault": [Function],
                      "_deps": Array [],
                      "_excludedEdges": Array [],
                      "_exclusive": Object {},
                      "_mutate": false,
                      "_nodes": Array [
                        "email",
                      ],
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "object",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "fields": Object {
                        "email": StringSchema {
                          "_blacklist": RefSet {
                            "list": Set {},
                            "refs": Map {},
                          },
                          "_conditions": Array [],
                          "_deps": Array [],
                          "_exclusive": Object {
                            "required": false,
                            "undefined": false,
                          },
                          "_mutate": false,
                          "_options": Object {
                            "abortEarly": true,
                            "recursive": true,
                          },
                          "_type": "string",
                          "_typeError": [Function],
                          "_whitelist": RefSet {
                            "list": Set {},
                            "refs": Map {},
                          },
                          "tests": Array [
                            [Function],
                            [Function],
                            [Function],
                          ],
                          "transforms": Array [
                            [Function],
                          ],
                        },
                      },
                      "tests": Array [],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "values": Object {
                      "email": "",
                    },
                  }
                }
                name="email"
              />
            </FormikConnect(ErrorMessageImpl)>
          </div>
          <button
            disabled={true}
            isRegistration={true}
            type="submit"
          >
            <button
              className="ParticipantButton undefined"
              disabled={true}
              type="submit"
            >
              registration.loginlink.button
            </button>
          </button>
        </form>
      </Component>
    </Form>
  </Formik>
</LoginLink>
`;
